sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -211,7 +211,6 @@ object Cast {
   }
 
   def forceNullable(from: DataType, to: DataType): Boolean = (from, to) match {
-    case (NullType, _) => true
```

Comment:
is it safe? What if we cast `array(null)` to `array<int>`?

Reply:
`ArrayType(NullType, containsNull=False)` can't contain `null`, no?

Reply:
Maybe we should clearly define what `forceNullable` should return.

From the name, seems like we should return true if `from = NullType`.

Reply:
and seems it replaces https://github.com/apache/spark/commit/d7b97a1d0daf65710317321490a833f696a46f21 ?

Reply:
Yeah .. let me remove that change here as well. Let me add some documentation there as well.